### PR TITLE
pref: optimize CPU mode analyze by switching from rustfft to realfft

### PR DESCRIPTION
**Pull request by XMLHexagram**
When testing startup_0.ogg in CPU mode on my m3 max MacBook , this PR can optimize analyze time from  approximately 37+ms -> 36+ms.

~~While working on this PR, I found  one way to speed up(also about 1ms) data preparation during analyze, which will be completed in the next PR.~~
I will do this optimization in the refactoring PR.

After all these, I will refactor the code under the analyze mod.

## Summary by Sourcery

Optimize CPU mode analysis by replacing rustfft with realfft, resulting in improved performance. Update tests to validate the accuracy and performance of the new FFT implementation against legacy and GPU methods.

Enhancements:
- Optimize CPU mode analysis by switching from rustfft to realfft, reducing analysis time by approximately 1ms.

Tests:
- Add new tests to compare the performance and accuracy of real FFT against legacy and GPU implementations, ensuring consistency across different FFT methods.

### CODE REVIEW

**Comment by Losses** [comment]
Please install [clippy](https://github.com/rust-lang/rust-clippy)

**Comment by XMLHexagram** [comment]
Before:
`test fft_processor::tests::bench_cpu_fft ... bench:  35,596,679.20 ns/iter (+/- 801,572.33)`
After:
`test fft_processor::tests::bench_cpu_fft ... bench:  35,014,545.80 ns/iter (+/- 2,090,121.57)`

File: analysis/src/fft_processor.rs [hunk 1]
```diff
@@ -409,24 +422,44 @@ impl FFTProcessor {
             .expect("CPU computing device not initialized");
 
         if force {
-            cpu_fft.process(&mut self.batch_fft_buffer);
+            let _ = cpu_fft.process(
```
[the remark this answers is not in the document]
Comment (XMLHexagram) [reply analysis/src/fft_processor.rs:425]: Can I apply to postpone this request to the refactoring PR? @Losses

**Comment by Losses** [comment]
While the optimization of analyze time from approximately 37+ms to 36ms is a step in the right direction, a 1ms improvement may not be particularly impressive. Please ensure that these **results were obtained in release mode**. Additionally, provide a [t-test report](https://en.wikipedia.org/wiki/Student%27s_t-test) to confirm that this result is not due to random variance but represents **a stable and significant optimization**. We require that the statistical results show a p-value of **less than 0.001**.

Furthermore, please report on the impact of this optimization on the compiled output. If there is a significant increase in the binary size, we may consider this optimization not worthwhile.

File: analysis/src/fft_processor.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (XMLHexagram) [reply analysis/src/fft_processor.rs]: fixed

File: analysis/src/fft_processor.rs [hunk 2]
```diff
@@ -409,24 +415,44 @@ impl FFTProcessor {
             .expect("CPU computing device not initialized");
 
         if force {
-            cpu_fft.process(&mut self.batch_fft_buffer);
+            cpu_fft.process(
```
[the remark this answers is not in the document]
Comment (XMLHexagram) [reply analysis/src/fft_processor.rs:418]: It's a good idea, I already do this in #69 before, but it's closed due to too many conflicts to the main branch.

I will reconsider and implement this in a refactoring PR in the future.

**Comment by XMLHexagram** [comment]
> Furthermore, please report on the impact of this optimization on the compiled output. If there is a significant increase in the binary size, we may consider this optimization not worthwhile.

This PR only import a very tiny library called `realfft` which is based on `rustfft`, so it won't significantly increase the binary size.

**Comment by XMLHexagram** [comment]
```
Running benches/analysis_benchmark.rs (target/release/deps/analysis_benchmark-e1c94062997c57be)
cpu analysis/cpu analysis
                        time:   [35.839 ms 35.866 ms 35.896 ms]
                        change: [-1.8871% -1.4666% -1.1097%] (p = 0.00 < 0.01)
                        Performance has improved.
Found 6 outliers among 100 measurements (6.00%)
  6 (6.00%) high mild
```

**Pull request merged**